woocommerce/woocommerce
# Add/42 new product page

### All Submissions:






### Changes proposed in this Pull Request:



This PR adds the code to:
- create a new feature flag for the new product management experience (`new-product-management-experience`).
- render a blank reactified page when the flag is enabled and the user navigates to Products > Add New.

![screenshot-clean local-2022 07 29-10_06_00](https://user-images.githubusercontent.com/1314156/181767935-f885b562-6c0b-4da0-9468-a562e3845be6.png)

- modify the href for the following buttons
 
`Products > Add New` (using the old and the new navigation) and the `Add New` button under `Products > All Products`.

![screenshot-clean local-2022 07 29-10_05_52](https://user-images.githubusercontent.com/1314156/181768048-aa60636d-5d4e-491d-ad11-8e510e3efaa1.png)

The `Add manually` in the `Add products` task.

![screenshot-clean local-2022 07 29-10_06_19](https://user-images.githubusercontent.com/1314156/181767814-45482303-efdf-4c92-82d6-5286725b5471.png)

Closes this issue (42-gh-woocommerce/mothra-private).

### UPDATE

Since there is no button to create a product with another type other than simple, the commit [2953fb1](https://github.com/woocommerce/woocommerce/pull/34115/commits/2953fb1dc558dbd7d21aff749080c031ad35345e) has been created to add a menu item named `Products` > `Add New (MVP)` and remove (temporarily) the URL override of the old buttons.

After having the button mentioned above in place, the commit [2953fb1](https://github.com/woocommerce/woocommerce/pull/34115/commits/2953fb1dc558dbd7d21aff749080c031ad35345e) should be deleted.

### How to test the changes in this Pull Request:

1. Checkout this branch.
2. Turn on the feature flag by using the plugin [woocommerce-admin-test-helper](https://github.com/woocommerce/woocommerce-admin-test-helper) (link [here](https://github.com/woocommerce/woocommerce-admin-test-helper/releases/download/v0.7.6/woocommerce-admin-test-helper.zip)). You will see the option `new-product-management-experience` under `Tools > WCA Test Helper > Features`.
3. Refresh the page.
4. Verify that a blank page is shown with the title `Add product` after pressing the following buttons:
`Products > Add New` and the `Add New` button under `Products > All Products` and the `Add manually` in the `Add my products` task.
5. Verify the event `view_new_product_management_experience` is being recorded when viewing the `Add product` page.
6. Verify that it works as expected using the new navigation too.
7. Turn off the feature flag and verify that the `Add New` button works as expected.

### Other information:




### FOR PR REVIEWER ONLY:
